sycl/test/extensions/properties/properties_kernel_device_has_warning.cpp

```diff
@@ -0,0 +1,167 @@
+// RUN: %clangxx -fsycl-device-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+// Tests for warnings when propagated aspects do not match the aspects available
+// in a function, as specified through the 'sycl::device_has' property.
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+
+// expected-note-re@+1 4 {{propagated from call to function '{{.*}}Struct1::Struct1({{.*}})'}}
+struct [[__sycl_detail__::__uses_aspects__(aspect::fp16)]] Struct1 {
```

Comment:
nit: I'd suggest renaming structs/funcs to something like `StructWithFP16`, `use_fp16`, `use_cpu`, etc. Not sure how much help it would be though.